WcaOnRails/app/models/regulation.rb

```diff
@@ -0,0 +1,16 @@
+class Regulation < SimpleDelegator
+  begin
+    REGULATIONS = JSON.parse(File.read(Rails.root.to_s + "/app/views/search_results/wca-regulations.json")).freeze
+  rescue
+    REGULATIONS = [].freeze
+  end
+
+  def limit(number)
+    first(number)
+  end
+
+  def self.search(query, *)
+    query.downcase!
+    Regulation.new(REGULATIONS.select { |reg| (reg['text'].downcase.include?(query) || reg['id'].downcase.include?(query)) })
```

Comment:
This can be like:
```ruby
matched_regulations = REGULATIONS
query.downcase.split do |part|
  matched_regulations.select! do |reg|
    %w(text id).any? { |field| reg[field].downcase.include?(part) }
  end
end
Regulation.new(matched_regulations)
```
I haven't tried this but it should work. As you have the JSON file it will be nice if you can just check in the console whether it does its job =)